WcaOnRails/app/models/competition.rb

```diff
@@ -331,6 +331,20 @@ def warnings_for(user)
       unless has_any_round_per_event? && schedule_includes_rounds?
         warnings[:schedule] = I18n.t('competitions.messages.schedule_must_match_rounds')
       end
+
+      warnings = championship_warnings.merge(warnings)
+    end
+
+    warnings
+  end
+
+  def championship_warnings
+    warnings = {}
+    for championship in self.championships
+      existing_championship = Championship.joins(:competition).where('Competitions.showAtAll AND championship_type = ? AND Competitions.year = ?', championship.championship_type, self.year)
+      if existing_championship.length > 0
+        warnings[championship.championship_type] = I18n.t('competitions.messages.championship_exists', championship_type: championship.name, year: self.year)
```

Comment:
Won't this look a bit saturated if a competition has multiple championship types that have been already held on that year? Although that would be an extremely unlikely case, maybe collecting all the championship types and then pass them all at once is is better.

Reply:
I agree this would look very saturated with multiple championship-already-in-use warnings, but as there have only been 7 competitions with two championships (4 WCs, 3 ACs), that seems extremely unlikely to me.  If someone adds 10 championships that already exist, and it looks bad in the admin view, that seems like it's their fault :)

Given that only one already-used championship is much more likely, I want to make sure the warning text sounds natural for that case:
"There is already a %{championship_type} in %{year}" was the best I could come up with, but it doesn't handle the multiple case.
"The following championships have already been used in %{year}: %{championship_types}" supports the multiple case, but is less natural in the single case

I'm open to other suggestions!